Integrations/src/main/java/io/deephaven/integrations/numpy/Java2NumpyCopy.java

```diff
@@ -497,23 +497,21 @@ public static void copyRand(final Table t, final long[] data, final int nRow, fi
         copyRand(t, nRow, nCol, rows, long.class, "(long)", (cs, k, idx) -> data[idx] = cs.getLong(k));
     }
 
-    //this is commented out so the byte[] method will be called. Both are 8 bit data structures the boolean cast will
-    //throw an error
-//    /**
-//     * Casts data to the desired type and copies a random selection of rows into a flattened 2D array.
-//     * This is useful for copying table data directly into numpy arrays.
-//     *
-//     * @param t    table to copy data from
-//     * @param data array to copy data into
-//     * @param nRow number of rows to copy; also the number of rows in <code>data</code>.
-//     * @param nCol number of table columns; also the number of columns in <code>data</code>.
-//     * @param rows indices of rows to copy.  Null causes rows to be randomly generated.
-//     */
-//    public static void copyRand(final Table t, final boolean[] data, final int nRow, final int nCol, final long[] rows) {
-//        assertCopyRandArgs(t, data == null ? -1 : data.length, nRow, nCol, rows);
-//        assert data != null;
-//        copyRand(t, nRow, nCol, rows, boolean.class, "(boolean)", (cs, k, idx) -> data[idx] = cs.getBoolean(k));
-//    }
+    /**
+     * Casts data to the desired type and copies a random selection of rows into a flattened 2D array.
+     * This is useful for copying table data directly into numpy arrays.
+     *
+     * @param t    table to copy data from
+     * @param data array to copy data into
+     * @param nRow number of rows to copy; also the number of rows in <code>data</code>.
+     * @param nCol number of table columns; also the number of columns in <code>data</code>.
+     * @param rows indices of rows to copy.  Null causes rows to be randomly generated.
+     */
+    public static void copyRand(final Table t, final boolean[] data, final int nRow, final int nCol, final long[] rows) {
+        assertCopyRandArgs(t, data == null ? -1 : data.length, nRow, nCol, rows);
+        assert data != null;
+        copyRand(t, nRow, nCol, rows, boolean.class, "(boolean)", (cs, k, idx) -> data[idx] = cs.getBoolean(k));
+    }
```

Comment:
This comment seems to indicate that these constructors were intentionally removed so that byte[] is used.  Does the boolean cast throw an error?  Are you certain this code is executed?

Reply:
Looking at the code, I think there is a test.  Possibly the concern is `Java2NumpyCopy.java:382`.  It looks like we automatically cast from type A to type B.  Maybe there is something funny there.  It isn't obvious to me what the problem is.  Just make certain that the unit tests are getting run.  You can also run them with code coverage in IJ to make sure that the code is getting executed.

Reply:
Tested with coverage and all of these methods are covered.